libusb.go

```diff
@@ -169,13 +173,37 @@ type libusbIntf interface {
 }
 
 // libusbImpl is an implementation of libusbIntf using real CGo-wrapped libusb.
-type libusbImpl struct{}
+type libusbImpl struct {
+	logLevel         int
+	disableDiscovery bool
+	useUsbDK         bool
+}
 
-func (libusbImpl) init() (*libusbContext, error) {
+func (impl libusbImpl) init() (*libusbContext, error) {
 	var ctx *C.libusb_context
-	if err := fromErrNo(C.libusb_init(&ctx)); err != nil {
+
+	var libusb_options [4]C.struct_libusb_init_option // fixed to 4 - there are maximum 4 options
+	n_options := 0
+	if impl.logLevel != C.LIBUSB_LOG_LEVEL_NONE {
+		libusb_options[n_options].option = C.LIBUSB_OPTION_LOG_LEVEL
+		b := make([]byte, 8)
+		binary.LittleEndian.PutUint64(b, uint64(impl.logLevel))
+		copy(libusb_options[n_options].value[:], b)
+		n_options += 1
+	}
+	if impl.useUsbDK {
```

Comment:
ditto here - I don't think this is needed for Android support? Move to a separate PR